boards: adafruit: add support for adafruit esp32s3 feather tft reverse

DBS06 opened the pull request:
This PR adds board support for the [Adafruit Feather ESP32S3 TFT Reverse](https://www.adafruit.com/product/5691) development board.

This is basically the last board missing from the Adafruit Feather ESP32S2 and ESP32S3 series, which were requested here #68512. In this issue there are also some other dev-boards listed, but are a different series (the naming is similar but are different EPS32 Modules) or revisions (this does effect only the ESP32S2 boards and are already supported by my last PR #85405). The Adafruit Feather ESP32S3 series does not have different revisions.

Furthermore I found a bug in the Adafruit Feather ESP32S3 TFT, which I thought should be corrected and also fixed in this PR.

I tested the following examples:

- `hello_world` (also with `--sysbuild`)
- `basic/blinky`
- `drivers/led/led_strip`
- `fuel_gauge/max17048` (obsolete when my other PR #90310 will be finally reviewed and merged)
- `subsys/display/lvgl`
- `subsys/console/echo`
- `subsys/shell/shell_module`
- `net/wifi/shell`

DBS06 committed:
boards: adafruit: esp32s3_feather_tft fix size of Flash and PSRAM

fix size of Flash and PSRAM

Signed-off-by: Philipp Steiner <philipp.steiner1987@gmail.com>

JarmouniA reviewed (changes requested):
This board can be simply added as a variant of `ESP32-S3 TFT Feather`
> It's basically our ESP32-S3 TFT Feather but with the 240x135 color TFT display on the back-side, not the front-side.

https://learn.adafruit.com/esp32-s3-reverse-tft-feather/overview

See https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#board-terminology

DBS06 commented on boards/adafruit/feather_esp32s3_tft_reverse/adafruit_feather_esp32s3_tft_reverse_appcpu.yaml:
```diff
@@ -0,0 +1,27 @@
+identifier: adafruit_feather_esp32s3_tft_reverse/esp32s3/appcpu
+name: Adafruit Feather ESP32-S3 TFT Reverse APPCPU
```
"Simply" is a strong word :smile: . Actually they differ in the pin assignment, which basically would lead nearly to the same amount of files in one folder, which are now in three folders.

All `Adafruit ESP32-S3 Feather` boards are and can be more or less a variant of each other. To be consistent with the already available `Adafruit ESP32-S3 Feather` boards I added the `Adafruit ESP32-S3 Feather Reverse TFT` board in its own folder.

If I would add it as a variant of `Adafruit ESP32-S3 Feather TFT` board, then it would be consistent to add both board as a variant of the `Adafruit ESP32-S3 Feather`, like I did it with the `Adafruit ESP32-S2 Feather` boards.

But nevertheless, I once discussed that with @kartben and he said that both ways are acceptable and there are no specific ways defined how to do that (I couldn't find the post again and I am unsure if it was on Github or on Discord).

In my opinion and from my experience it is easier to add and maintain the board variants in different folders and adding all variants in the same folder it does not reduce the amount of code duplication as someone would expect.

Personally I don't have a preference which way to do it, but if I should add the `Adafruit ESP32-S3 Feather Reverse TFT` as a variant of `Adafruit ESP32-S3 Feather TFT` and/or add both boards as a variant of `Adafruit ESP32-S3 Feather` someone needs to send me a sample of the `Adafruit ESP32-S3 Feather` and  `Adafruit ESP32-S3 Feather TFT` to test it properly or does the testing for me.
During developing/adding the `Adafruit ESP32-S2 Feather` boards, I had the luck to find someone who had the `Adafruit ESP32-S2 Feather Rev B` and `Rev C` boards and was willing to test it for me, but this wasn't easy and without proper testing I can't guarantee that the boards are working properly.

DBS06 committed:
boards: adafruit: add initial support esp32s3_feather_tft_reverse

The Adafruit Feather ESP32S3 TFT Reverse is a development board in the
Feather standard layout, sharing peripheral placement with other devices
labeled as Feathers or FeatherWings. The board is equipped with an
ESP32-S3 mini module, a fuel gauge, a USB-C and Qwiic/STEMMA-QT connector.
This variant additionally comes with a 240x135 pixel IPS TFT color display
on the backside of the boards and with 3 buttons.

Signed-off-by: Philipp Steiner <philipp.steiner1987@gmail.com>

DBS06 commented:
> This board can be simply added as a variant of `ESP32-S3 TFT Feather`
> 
> > It's basically our ESP32-S3 TFT Feather but with the 240x135 color TFT display on the back-side, not the front-side.
> 
> https://learn.adafruit.com/esp32-s3-reverse-tft-feather/overview
> 
> See https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#board-terminology
